app/repository/ReliefsMongoRepository.scala

```diff
@@ -78,13 +79,15 @@ class ReliefsReactiveMongoRepository(mongo: () => DB, val metrics: ServiceMetric
 
   def updateTimeStamp(relief: ReliefsTaxAvoidance, date: DateTime): Future[ReliefCached] = {
     val query = BSONDocument("periodKey" -> relief.periodKey, "atedRefNo" -> relief.atedRefNo)
-    collection.update(ordered = false).one(query, relief.copy(timeStamp = date), upsert = false, multi = false) map { res =>
-      if (res.ok && res.nModified != 0) {
-        logger.info(s"[updateTimestamp] Updated timestamp for ${relief.atedRefNo} with $date")
-        ReliefCached
-      } else {
-        logger.error(s"[updateTimeStamp: Relief] Mongo failed to update, problem occurred in collect - ex: $res")
-        ReliefCachedError
+    Mdc.preservingMdc {
```

Comment:
```
    Mdc.preservingMdc (collection.update(ordered = false).one(query, relief.copy(timeStamp = date), upsert = false, multi = false)) map { res =>
```
I.e. only wrap the mongo call if we want the subsequent `logger.info` to have had the MDC restored.